website/main/content/blog/2025-04-14-data-availability.md

```diff
@@ -0,0 +1,142 @@
+  that can be used to verify them in the upper layers of the system.
+- _Recursive operation (L1 Child):_ The protocol is recursive, so in the same way that the L1's
+  child shard was committing blocks to the parent as they were being created, it will do the same
+  with its own parent (the beacon chain).
+- _Super-Segment Commitment (L1 Child):_ When a super-segment from a child (`super_segment_L2_1`) is
+  committed, it is added to the shard's history buffer along with other local segments that may have
+  been created, triggering the clearing of the block buffer for child shards. Once a segment is
+  committed in the parent, there is no need to keep raw blocks in the buffer anymore as they are
+  explicitly available in the committed segments.
+- _Beacon Chain Integration (L1 Child):_ The beacon chain receives the parent shard's blocks and
+  segments, committing them to its chain (as it would happen with any regular parent shard in the
+  system).
+- _Super-Segment Commitment (L1 Child):_ When a super-segment from a child (`super_segment_L1_1`) is
+  committed, it's added to the shard's history buffer, triggering the clearing of the block buffer.
+- _Segment History (L1 Child):_ Segments in a child shard are created as super-segments, committed
+  with a list of shard and child super-segments created up to that point.
+- _Beacon Chain Role:_ The beacon chain commits all super-segments (`segment_commitment_L1_1`,
+  `super_segment_BC_1`, etc.), maintaining the whole system's historical buffer and creating a
+  unified history.
+
+<p align="center">
+<img alt="High-level stages and primitives for sharded archiving" src="2025-04-14-sharded-archiving.png"></img>
+</p>
+
+## How many shards can we afford if we are committing every block?
+
+You see in the description from the previous section that one of the key design changes since last
+week is that we will be committing every block to the parent to have part of the history of child
+shards as raw blocks until full segments for the shard are created.
+
+Assuming that we dedicate only a 10% of the full size of a block for the commitment of child blocks,
+let's do a back-of-the-envelope calculation of the total number of child shards that we can afford.
+Assuming:
+
+- `BLOCK_SIZE = 4MiB`
+- `BLOCK_TIME = 6 seconds`
+- `BLOCK_HEADER = 224B`
+
+```
+type BlockHeader (260B) {
+    number (8B)
+    extrinsic_root (32B)
+    state_root (32B)
+    parent_hash (32B)
+    shard_id (4B)
+    hash (32B)
+    solution (120B)
+}
+```
+
+And that every 6 seconds all the shards in the lower level submit a block at the same time (this is
+the worst case scenario), we can support the following number of shards in the lower level:
+
+```
+NUM_SHARDS = (0.10 * BLOCK_SIZE) / BLOCK_HEADER =~ 153 SHARDS
+```
+
+This assumes that the block time of shards is also 6 seconds and that the worst case scenario
+happens where all shards submit their blocks at the same time on a block.
```

Comment:
This is actually an average case, not worst case. It is much worse when block production happens later on a parent shard and we have more than one block to include from each child (though this depends on how we decide to do this, we might only include the last one here, in which case what you have described is the worst case indeed).

Reply:
I see your point. Let's consider this the "average scenario" for now (although I think that the numbers won't change much even in the worst case scenario, because we have at one block per slot per shard, which means that may skew a few blocks up and down, but it won't be relevant in terms of block space required asymptotically).